test/credentials_test.dart

```diff
@@ -299,4 +299,185 @@ Future<void> main([List<String>? args]) async {
+  ///    "name": {
+  ///      "firstName": "John",
+  ///      "lastName": "Doe"
+  ///    },
+  ///    "email": "JWK_URI_validated_user@realm.io",
+  ///    "gender": "male",
+  ///    "birthDay": "1999-10-11",
+  ///    "minAge": "10",
+  ///    "maxAge": "90",
+  ///    "company": "Realm",
+  ///    "iat": 1660130241,
+  ///    "exp": 1660349241, //10 years after Aug 2022
+  ///    "aud": "mongodb.com",
+  ///    "iss": "https://realm.io"
+  /// }
+  /// JWKS url validation is configured in autoConfirm app.
+  baasTest('JWT validation by using JWK URI - login', (configuration) async {
+    final app = App(configuration);
+    String username = "JWK_URI_validated_user@realm.io";
+    String userId = "62f39187e4b27ad3fb6483ca";
+    var token =
+        "eyJraWQiOiIxIiwiYWxnIjoiUlMyNTYiLCJ0eXAiOiJKV1QifQ.eyJzdWIiOiI2MmYzOTE4N2U0YjI3YWQzZmI2NDgzY2EiLCJuYW1lIjp7ImZpcnN0TmFtZSI6IkpvaG4iLCJsYXN0TmFtZSI6IkRvZSJ9LCJlbWFpbCI6IkpXS19VUklfdmFsaWRhdGVkX3VzZXJAcmVhbG0uaW8iLCJnZW5kZXIiOiJtYWxlIiwiYmlydGhEYXkiOiIxOTk5LTEwLTExIiwibWluQWdlIjoiMTAiLCJtYXhBZ2UiOiI5MCIsImNvbXBhbnkiOiJSZWFsbSIsImlhdCI6MTY2MDEzMDI0MSwiZXhwIjoxNjYwMzQ5MjQxLCJhdWQiOiJtb25nb2RiLmNvbSIsImlzcyI6Imh0dHBzOi8vcmVhbG0uaW8ifQ.KXpndzjQq2v0K2jLygNU6574VTFQvdyQQH6sEjlDik-ZM7IGVPHZb-o8PWKaMK1DH8V8JEG1fi7Nf5CDHk5XWDgBVcrGfB2qrMq_MZpdPvuSvKbnfN9ZQBME58l9XK-_2RszEPiULy3u_daYCwitNOGFhJ5PJUhC487CNsi3gK947hMMwBDhnF4CcQ68EEE5IRFNcvLbnrDZhKKceYbt0YhGmjTBd-Y0DKF0mO3wUTk-Y4xeXcxd_iVZr-uMa5xbjdsYbI9EwWMmTs6RvW4hWqrAXaAkBbnhrIl_dbGCx6hmCAVtRPNDu7R1H63svITfAgNrzSEDeeIROnU0e9IA4w";
+    final credentials = Credentials.jwt(token);
+    final user = await app.logIn(credentials);
+
+    expect(user.state, UserState.loggedIn);
+    expect(user.identities[0].id, userId);
+    expect(user.provider, AuthProviderType.jwt);
+    expect(user.profile.email, username);
+    expect(user.profile.name, username);
+    expect(user.profile.gender, "male");
+    expect(user.profile.birthDay, "1999-10-11");
+    expect(user.profile.minAge, "10");
+    expect(user.profile.maxAge, "90");
+    expect(user.profile.firstName, "John");
+    expect(user.profile.lastName, "Doe");
+    expect(user.profile["company"], "Realm");
+  }, appName: AppNames.autoConfirm);
+
+/// JWT Header:
+  /// {
+  ///   "kid": "1",
+  ///   "alg": "RS256",
+  ///   "typ": "JWT"
+  /// }
+  /// JWT Payload data
+  /// {
+  ///   "sub": "62f3840b4ac43f38a50b9e2b", //User.identities.id of the existing user realm-test@realm.io
+  ///   "name": {
+  ///     "firstName": "John",
+  ///     "lastName": "Doe"
+  ///   },
+  ///   "email": "realm-test@realm.io",
+  ///   "gender": "male",
+  ///   "birthDay": "1999-10-11",
+  ///   "minAge": "10",
+  ///   "maxAge": "90",
+  ///   "company": "Realm",
+  ///   "iat": 1660131880,
+  ///   "exp": 1660350880, //10 years after Aug 2022
+  ///   "aud": "mongodb.com",
+  ///   "iss": "https://realm.io"
+  /// }
+  baasTest('JWT - login with existing user and edit profile', (configuration) async {
```

Comment:
This test can't work like this. Users are distinguished by provider+providerId, so an email password user cannot authenticate with JWT, even if their identities are the same. What you can do is generate two tokens for the same user identity and different profile information. Then validate that both authenticate the same user and that after the second auth, the profile has been updated.

Reply:
I'm just rewriting this test with having one user with two identities for different providers.

Reply:
Does it make sense now?